File: bakery-inventory-backend/src/main/java/com/bakery/inventory/service/CategoryService.java
package com.bakery.inventory.service;

import com.bakery.inventory.dto.category.CategoryRequest;
import com.bakery.inventory.dto.category.CategoryResponse;

import java.util.List;

public interface CategoryService {

    CategoryResponse createCategory(CategoryRequest request);

    List<CategoryResponse> getAllCategories();

    CategoryResponse getCategoryById(Integer id);

    CategoryResponse updateCategory(Integer id, CategoryRequest request);

    void deleteCategory(Integer id);
}

File: bakery-inventory-frontend/src/services/categoryService.js
import axiosInstance from '../api/axiosInstance';

/**
 * NEW FILE: CategoryService
 * Manages category browsing and Admin CRUD operations.
 */

export const categoryService = {
  getAllCategories: async () => {
    const response = await axiosInstance.get('/api/categories');
    return response.data;
  },

  getCategoryById: async (id) => {
    const response = await axiosInstance.get(`/api/categories/${id}`);
    return response.data;
  },

  createCategory: async (categoryData) => {
    // categoryData = { name, description }
    const response = await axiosInstance.post('/api/categories', categoryData);
    return response.data;
  },

  updateCategory: async (id, categoryData) => {
    const response = await axiosInstance.put(`/api/categories/${id}`, categoryData);
    return response.data;
  },

  deleteCategory: async (id) => {
    const response = await axiosInstance.delete(`/api/categories/${id}`);
    return response.data;
  }
};
